### PR TITLE
Bug 1990384: Improve Alertmanager unavailable message

### PR DESCRIPTION
**Pull request by jgbernalp**
Add an alert suggesting that the alertmanager may be unavailable

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=199038

Screenshot 2022-04-26 at 12 08 44

### CODE REVIEW

**Comment by jgbernalp** [comment]
@juzhao The Alertmanager manges silences when it's disabled silences won't load. Currently there is no other api that can load the silences in this situation. The error can arise when the Alertmanager is actually unavailable due to errors or when it was disabled by the user, from the console point of view this is indistinguishable. Letting the console know when the user manually disables the Alertmanager will require exposing the configuration, this effort won't be reflected on user satisfaction, as most likely if the user disabled the Alertmanager it won't use the Alerts section on the console.

File: frontend/public/components/monitoring/alerting.tsx [hunk 1]
```diff
@@ -1380,6 +1380,18 @@ const MonitoringListPage: React.FC<ListPageProps & {
             {silencesLoadError.json?.error || silencesLoadError.message}
           </PFAlert>
         )}
+        {loadError && (
```
Comment (kyoto) [inline frontend/public/components/monitoring/alerting.tsx:1383]: `loadError` can be the load error for the alerts, the rules or the silences depending on the page.

We already have `silencesLoadError` available, so I think we just need to use that. In other words, I think we just need to change the `silencesLoadError && !loadError` case above.

**Review by kyoto — CHANGES_REQUESTED** [review]

File: frontend/public/components/monitoring/alerting.tsx (hunk 1 above)
Comment (jgbernalp) [reply frontend/public/components/monitoring/alerting.tsx]: Good catch @kyoto . I realized the `loadError` in some cases is the same value as `silencesLoadError`, in particular when you are in the silences tab. I refactored the code to lift the Alert component so the consumers of the `MonitoringListPage` component are able to decide which alert to display. WDYT? 👀

File: frontend/public/components/monitoring/types.ts [hunk 2]
```diff
@@ -50,19 +50,19 @@ export type MonitoringResource = {
 export type Silences = {
   data: Silence[];
   loaded: boolean;
-  loadError?: string;
+  loadError?: any;
```
Comment (kyoto) [inline frontend/public/components/monitoring/types.ts:53]: I thought `loadError` was always a `string`?

File: contrib/oc-environment.sh [hunk 3]
```diff
@@ -25,10 +25,12 @@ export BRIDGE_K8S_MODE_OFF_CLUSTER_ENDPOINT
 BRIDGE_K8S_MODE_OFF_CLUSTER_SKIP_VERIFY_TLS=true
 export BRIDGE_K8S_MODE_OFF_CLUSTER_SKIP_VERIFY_TLS
 
-BRIDGE_K8S_MODE_OFF_CLUSTER_THANOS=$(oc -n openshift-config-managed get configmap monitoring-shared-config -o jsonpath='{.data.thanosPublicURL}')
+CLUSTER_THANOS=$(oc -n openshift-config-managed get configmap monitoring-shared-config -o jsonpath='{.data.thanosPublicURL}')
```
Comment (kyoto) [inline contrib/oc-environment.sh:28]: I guess this was included by accident

**Review by kyoto — CHANGES_REQUESTED** [review]
Thanks @jgbernalp. This seems like a good solution to me.

File: frontend/public/components/monitoring/alerting.tsx (hunk 1 above)
Comment (kyoto) [reply frontend/public/components/monitoring/alerting.tsx]: 💯

File: contrib/oc-environment.sh (hunk 3 above)
Comment (jgbernalp) [reply contrib/oc-environment.sh]: yes, I just removed it

File: frontend/public/components/monitoring/types.ts (hunk 2 above)
Comment (jgbernalp) [reply frontend/public/components/monitoring/types.ts:53]: Is passed into the action as an error directly from the request.
https://github.com/openshift/console/blob/master/frontend/public/actions/observe.ts#L78

Comment (kyoto) [reply frontend/public/components/monitoring/types.ts:53]: Could we make it `loadError?: string | Error`?

**Pull request merged**